### PR TITLE
Document Mono-specific classes

### PR DESCRIPTION
**Pull request by Calinou**
This fully documents all 3 Mono-specific classes.

### CODE REVIEW

File: modules/mono/doc_classes/GodotSharp.xml [hunk 1]
```diff
@@ -11,26 +14,30 @@
...
-				Detaches the current thread from the mono runtime.
+				Detaches the current thread from the Mono runtime.
 			</description>
 		</method>
 		<method name="get_domain_id">
 			<return type="int">
 			</return>
 			<description>
+				Returns the engine MonoDomain's ID.
+				[b]Note:[/b] The Mono runtime must be initialized for this method to work (use [method is_runtime_initialized] to check). If the Mono runtime isn't initialized at the time this method is called, the engine will crash.
```
Comment (akien-mga) [inline modules/mono/doc_classes/GodotSharp.xml:32]: It's weird that calling a getter would crash the engine.
@neikeq I'd suggest changing this to a simple `ERR_FAIL_COND`. (Same for other occurrences.)

The engine should *not* crash when using the public scripting API.

Comment (Calinou) [reply modules/mono/doc_classes/GodotSharp.xml:32]: Since this method returns an integer, should it return `-1` in case an error occurs?

Comment (akien-mga) [reply modules/mono/doc_classes/GodotSharp.xml:32]: I guess, as long as it's not a valid MonoDomain ID.

File: modules/mono/doc_classes/GodotSharp.xml [hunk 2]
```diff
@@ -11,26 +14,30 @@
...
 			<description>
-				Detaches the current thread from the mono runtime.
+				Detaches the current thread from the Mono runtime.
 			</description>
 		</method>
 		<method name="get_domain_id">
 			<return type="int">
 			</return>
 			<description>
+				Returns the engine MonoDomain's ID.
```
Comment (neikeq) [inline modules/mono/doc_classes/GodotSharp.xml:31]: It returns the current domain.

File: modules/mono/doc_classes/GodotSharp.xml [hunk 3]
```diff
@@ -11,26 +14,30 @@
...
+				[b]Note:[/b] The Mono runtime must be initialized for this method to work (use [method is_runtime_initialized] to check). If the Mono runtime isn't initialized at the time this method is called, the engine will crash.
 			</description>
 		</method>
 		<method name="get_scripts_domain_id">
 			<return type="int">
 			</return>
 			<description>
+				Returns the scripts MonoDomain's ID. On most platforms, this will return a different MonoDomain ID from [method get_domain_id]. However, on non-simulator iOS builds, it will return the same MonoDomain ID as static ahead-of-time compilation is used there due to platform limitations.
```
Comment (neikeq) [inline modules/mono/doc_classes/GodotSharp.xml:39]: `get_domain_id` should actually return the same as `get_scripts_domain_id` most of the time, unless it's not loaded.

**Comment by neikeq** [comment]
Would be great if we could get rid of the `@C#` doc class as it serves no purpose. Maybe discard it if it's empty.

**Comment by akien-mga** [comment]
> Would be great if we could get rid of the `@C#` doc class as it serves no purpose. Maybe discard it if it's empty.

Done with #39622.

**Review by akien-mga — APPROVED** [review]
We can handle the crashing methods in a separate issue/PR, the documentation is accurate.

**Comment by akien-mga** [comment]
Thanks!

File: doc/classes/Texture.xml [hunk 4]
```diff
@@ -1,10 +1,14 @@
 <?xml version="1.0" encoding="UTF-8" ?>
 <class name="Texture" inherits="Resource" version="4.0">
```
Comment (akien-mga) [inline doc/classes/Texture.xml:2]: Welp... That was a bogus rebase :)

File: doc/classes/Texture.xml [hunk 5]
```diff
@@ -1,10 +1,14 @@
...
 	<brief_description>
+		The global C# namespace (Mono-enabled builds only).
```
Comment (akien-mga) [inline doc/classes/Texture.xml:4]: Welp... That was a bogus rebase :)

Comment (akien-mga) [reply doc/classes/Texture.xml:4]: Fixed in dfa5f88c11.

**Comment by akien-mga** [comment]
Cherry-picked for 3.2.2. (Without the `Texture` changes reverted in dfa5f88c11.)

File: doc/classes/Texture.xml (hunk 5 above)
Comment (Calinou) [reply doc/classes/Texture.xml:4]: Sorry, I didn't notice this in the rebase log. I was wondering where `@C#.xml` went.

**Pull request merged**